mne/decoding/gsoc.py

```diff
@@ -0,0 +1,249 @@
+"""Contains refactored transformers compatible with scikit-learn"""
+
+import numpy as np
+from scipy import linalg
+
+from .. import Covariance
+from ..cov import _regularized_covariance
+from ..preprocessing.xdawn import Xdawn
+
+
+class XdawnTransformer(Xdawn):
+
+    """Implementation of the Xdawn Algorithm.
+
+    Xdawn is a spatial filtering method designed to improve the signal
+    to signal + noise ratio (SSNR) of the ERP responses. Xdawn was originally
+    designed for P300 evoked potential by enhancing the target response with
+    respect to the non-target response. This implementation is a light version
+    of the original version. This does not handle overlap cases however it is
+    helpful while pipelining.
+
```

Comment:
Add an explicit description indicating that it doesn't take overlap into account